doc/lua_api.md

```diff
@@ -7472,6 +7472,19 @@ child will follow movement and rotation of that bone.
                 abides by, `"custom"` uses `sun_tint` and `moon_tint`, while
                 `"default"` uses the classic Minetest sun and moon tinting.
                 Will use tonemaps, if set to `"default"`. (default: `"default"`)
+        * `fog`: A table with following optional fields:
+            * `fog_distance`: integer, override the client's viewing_range.
+               By default, fog_distance is controlled by the client's viewing_range, and this field is not set.
+               When set the client client can no longer set a larger viewing_range (including range_all).
+               Any value >= 0 sets the desired fog_distance.
+               Any value < 0, resets the behavior to being client-controlled.
+               (default: not set)
+            * `fog_start`: float, override the client's fog_start.
+               Fraction of the visible distance at which fog starts to be rendered.
+               By default, fog_start is controlled by the client's `fog_start` setting, and this field is not set.
+               Any value between [0.0, 0.99] set the fog_start as a fraction of fog_distance.
```

Comment:
That's what the client setting has (see settingtypes). It's can be 100 and more than 99 does not make sense.
See `float fogShadingParameter = 1.0 / (1.0 - fogStart);`

Reply:
The 0.99 is only to prevent a division by 0, right? But 0.99 is pretty arbitrary, it could also be 0.999, for example. My point is that it's kinda ugly to expose this to the API. Can't we just take `[0.0, 1.0]` here, and hide the clamp as implementation detail in the client?